File: implementations/swift/RunTests.swift
// Test runner for Swift GA implementation
// Runs 25 instances and outputs timing results

import Foundation

struct RunTests {
    /// Run the GA benchmark multiple times and collect results
    static func runTests(numRuns: Int = 25) -> [Double] {
        print("Swift One-Max GA Performance Test")
        print("Running \(numRuns) tests...")
        
        var times: [Double] = []
        
        for i in 0..<numRuns {
            let elapsed = OneMaxGA.benchmarkSingleRun()
            times.append(elapsed)
            print("Run \(i + 1): \(String(format: "%.3f", elapsed)) ms", terminator: "\r")
            fflush(stdout)
        }
        
        print("\nCompleted \(numRuns) runs")
        
        // Output results in CSV format
        let timesStr = times.map { String($0) }.joined(separator: ",")
        print("swift,\(timesStr)")
        
        return times
    }
}

// Main execution
@main
struct Main {
    static func main() {
        RunTests.runTests(numRuns: 25)
    }
}

File: implementations/java/RunTests.java
// Test runner for Java GA implementation
// Runs 25 instances and outputs timing results

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class RunTests {
    
    /**
     * Run the GA benchmark multiple times and collect results.
     */
    public static List<Double> runTests(int numRuns) {
        System.out.println("Java One-Max GA Performance Test");
        System.out.println("Running " + numRuns + " tests...");
        
        List<Double> times = new ArrayList<>();
        
        for (int i = 0; i < numRuns; i++) {
            double elapsed = OneMaxGA.benchmarkSingleRun();
            times.add(elapsed);
            System.out.print("Run " + (i + 1) + ": " + String.format("%.3f", elapsed) + " ms\r");
            System.out.flush();
        }
        
        System.out.println("\nCompleted " + numRuns + " runs");
        
        // Output results in CSV format
        String timesStr = times.stream()
                .map(String::valueOf)
                .collect(Collectors.joining(","));
        System.out.println("java," + timesStr);
        
        return times;
    }
    
    public static void main(String[] args) {
        runTests(25);
    }
}